src/content/docs/cheerp/12-reference/05-sections/01-genericjs/index.md

```diff
@@ -0,0 +1,17 @@
+---
+title: genericjs
+description: The JavaScript section
+---
+
+Function compiled into the genericjs section will be output as JavaScript.
+
+To place functions in the genericjs section, you can:
+
+- Tag individual functions with the `[[cheerp::genericjs]]` attribute.
+- Use `-target cheerp` to default to genericjs for all functions.
+
+### Restrictions
+
+- Cannot do type unsafe things
+  - Converting a pointer to an integer then using it is undefined behaviour
+  - Cannot convert pointers between eachother. For example, you can't read `int*` as `float*` - this will give garbage. Instead, use a union of an int and a float.
```

Comment:
Maybe I'm being overly pedantic, but reading from an `int*` cast to `float*` is undefined in standard C++, and the value of a pointer cast to integer (even `uintptr_t`) is implementation-defined. Saying you cannot do these in genericjs doesn't make much sense to me, because you can't (or at least, shouldn't) do these things in "normal" C++ either.

The standard way of doing type punning is using `memcpy`, or (in C++20) `std::bit_cast`, of course these functions do not work in genericjs.

Maybe this section could use an example of a standard conforming C++ program that does not behave as expected using the genericjs target, and explain how and why the result is different.